terraform/groups/provisioning-role/module-iam-policies/terraform-apply.tf

```diff
@@ -0,0 +1,34 @@
+data "aws_iam_policy_document" "apply" {
+  statement {
+    sid       = "ApplyPolicyAllResources"
+    effect    = "Allow"
+    resources = ["*"]
```

Comment:
This could be restricted to only allow creation of resources named `payment-reconciler-*` to prevent rogue naming

Reply:
It is potentially also possible (needs spiking) to have an account wide default policy, which uses policy variables. Any role using this policy would then interpolate the variable as the name of the role using it, restricting the actions to only allowed on resources named consistently as the role. This would remove the need to create a policy per service

Reply:
I'm not sure it can for these ones. A lot of these (if you try to add them in the console) show the message **_Resources: The actions you chose support all resources_**
If you try to force it via JSON, you get the error: **_The actions in your policy do not support resource-level permissions and require you to chose All Resources_**.

I've defaulted to restricted resources where I can. Others have to be *, but I've added request conditions where the action has a suitable condition key - but many do not.
It's a rabbit hole, mate.

Reply:
I'd be interested in the variable thing. I did go looking for something similar - but could only see default / built-in variables (such as time, source IP, username, etc.). If there's a way to pass through a variable then that could be interesting. Then again it this is a 'policy' and you're passing something through - that might just be like using a wildcard in the policy and accepting anything. Probably worth a chat - easier than typing.